drivers/mbox/mbox_ti_omap.c

```diff
@@ -0,0 +1,230 @@
+					  mbox_callback_t cb, void *user_data)
+{
+	struct omap_mailbox_data *data = DEV_DATA(dev);
+	k_spinlock_key_t key;
+
+	if (channel >= MAILBOX_MAX_CHANNELS) {
+		return -EINVAL;
+	}
+
+	key = k_spin_lock(&data->lock);
+	data->cb[channel] = cb;
+	data->user_data[channel] = user_data;
+	k_spin_unlock(&data->lock, key);
+
+	return 0;
+}
+
+static int omap_mailbox_mtu_get(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return MAILBOX_MBOX_SIZE;
+}
+
+static uint32_t omap_mailbox_max_channels_get(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return MAILBOX_MAX_CHANNELS;
+}
+
+static int omap_mailbox_set_enabled(const struct device *dev, uint32_t channel, bool enable)
+{
+	const struct omap_mailbox_config *cfg = DEV_CFG(dev);
+	struct omap_mailbox_data *data = DEV_DATA(dev);
+	volatile struct omap_mailbox_regs *regs;
+	k_spinlock_key_t key;
+	uint32_t irqstatus;
+
+	if (channel >= MAILBOX_MAX_CHANNELS) {
+		return -EINVAL;
+	}
+
+	if (enable && data->channel_enable[channel]) {
+		return -EALREADY;
+	}
+
+	key = k_spin_lock(&data->lock);
+	regs = DEV_REG_BASE(dev);
+	irqstatus = regs->irq_regs[cfg->usr_id].enable_set;
+
+	if (enable) {
+		irqstatus |= MAILBOX_IRQ_NEWMSG(channel);
+	} else {
+		irqstatus &= ~MAILBOX_IRQ_NEWMSG(channel);
+	}
+
+	regs->irq_regs[cfg->usr_id].enable_set = irqstatus;
+	data->channel_enable[channel] = enable;
+
+	if (enable) {
+		irq_enable(cfg->irq);
+	} else {
+		irq_disable(cfg->irq);
+	}
+
+	k_spin_unlock(&data->lock, key);
+
+	return 0;
+}
+
+static const struct mbox_driver_api omap_mailbox_driver_api = {
+	.send = omap_mailbox_send,
+	.register_callback = omap_mailbox_register_callback,
+	.mtu_get = omap_mailbox_mtu_get,
+	.max_channels_get = omap_mailbox_max_channels_get,
+	.set_enabled = omap_mailbox_set_enabled,
+};
+
+#define MAILBOX_INSTANCE_DEFINE(idx)                                                               \
+	static struct omap_mailbox_data omap_mailbox_##idx##_data;                                 \
+	const static struct omap_mailbox_config omap_mailbox_##idx##_config = {                    \
+		DEVICE_MMIO_NAMED_ROM_INIT(reg_base, DT_DRV_INST(idx)),                            \
+		.irq = DT_INST_IRQN(idx),                                                          \
+		.usr_id = DT_INST_PROP(idx, usr_id),                                               \
+	};                                                                                         \
+	static int omap_mailbox_##idx##_init(const struct device *dev)                             \
+	{                                                                                          \
+		DEVICE_MMIO_NAMED_MAP(dev, reg_base, K_MEM_CACHE_NONE);                            \
+		IRQ_CONNECT(DT_INST_IRQN(idx), DT_INST_IRQ(idx, priority), omap_mailbox_isr,       \
+			    DEVICE_DT_INST_GET(idx),                                               \
+			    COND_CODE_1(DT_INST_IRQ_HAS_CELL(idx, flags),		\
+			    (DT_INST_IRQ(idx, flags)), (0)));       \
```

Comment:
Indention of the trailing `\` is off. I would also indent the second line like this:
```
COND_CODE_1(DT_INST_IRQ_HAS_CELL(idx, flags),
            (DT_INST_IRQ(idx, flags)), (0))); 
```